<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>593</width>
    <height>559</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>数据导入工具</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1071</width>
      <height>601</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>3</number>
    </property>
    <widget class="QWidget" name="tab_4">
     <attribute name="title">
      <string>导入设置</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>551</width>
        <height>131</height>
       </rect>
      </property>
      <property name="title">
       <string>数据源设置</string>
      </property>
      <widget class="QWidget" name="layoutWidget_3">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>90</y>
         <width>521</width>
         <height>25</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="1" colspan="2">
         <widget class="QLineEdit" name="tdx_dir_lineEdit"/>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>通达信安装目录：</string>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QPushButton" name="select_tdx_dir_pushButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>选择</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QRadioButton" name="tdx_radioButton">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>70</y>
         <width>481</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>使用通达信盘后数据（不支持分笔、分时数据，需要配置安装路径）</string>
       </property>
      </widget>
      <widget class="QWidget" name="layoutWidget_4">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>411</width>
         <height>22</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QRadioButton" name="pytdx_radioButton">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>使用Pytdx（推荐）</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_16">
          <property name="text">
           <string>同时使用</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="use_tdx_number_spinBox">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>10</number>
          </property>
          <property name="value">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_17">
          <property name="text">
           <string>个通达信服务器进行下载</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_7">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>170</y>
        <width>551</width>
        <height>261</height>
       </rect>
      </property>
      <property name="title">
       <string>导入设置</string>
      </property>
      <widget class="QWidget" name="layoutWidget_5">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>60</y>
         <width>155</width>
         <height>18</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QCheckBox" name="import_stock_checkBox">
          <property name="text">
           <string>股票</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="import_fund_checkBox">
          <property name="text">
           <string>基金</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="import_future_checkBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>期货</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="layoutWidget_6">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>90</y>
         <width>301</width>
         <height>18</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QCheckBox" name="import_day_checkBox">
          <property name="text">
           <string>日线</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="import_min5_checkBox">
          <property name="text">
           <string>5分钟线</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="import_min_checkBox">
          <property name="text">
           <string>1分钟线</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="import_trans_checkBox">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>分笔</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="import_time_checkBox">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>分时</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="layoutWidget_7">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>120</y>
         <width>271</width>
         <height>126</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="2" column="0">
         <widget class="QLabel" name="label_15">
          <property name="text">
           <string>初次导入1分钟线起始日期：</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>初次导入5分钟线起始日期：</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>初次导入日线起始日期：</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QDateEdit" name="min_start_dateEdit">
          <property name="calendarPopup">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QDateEdit" name="min5_start_dateEdit">
          <property name="minimumDateTime">
           <datetime>
            <hour>0</hour>
            <minute>0</minute>
            <second>0</second>
            <year>1990</year>
            <month>1</month>
            <day>1</day>
           </datetime>
          </property>
          <property name="calendarPopup">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>初次导入分时起始日期：</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>初次导入分笔起始日期：</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QDateEdit" name="day_start_dateEdit">
          <property name="minimumDateTime">
           <datetime>
            <hour>0</hour>
            <minute>0</minute>
            <second>0</second>
            <year>1990</year>
            <month>1</month>
            <day>1</day>
           </datetime>
          </property>
          <property name="calendarPopup">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QDateEdit" name="trans_start_dateEdit">
          <property name="calendarPopup">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QDateEdit" name="time_start_dateEdit">
          <property name="calendarPopup">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="layoutWidget_8">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>198</width>
         <height>18</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QCheckBox" name="import_weight_checkBox">
          <property name="text">
           <string>下载权息数据</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>存储设置</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>240</y>
        <width>551</width>
        <height>241</height>
       </rect>
      </property>
      <property name="title">
       <string>MYSQL存储设置</string>
      </property>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>80</y>
         <width>421</width>
         <height>131</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0">
         <widget class="QLabel" name="label_18">
          <property name="text">
           <string>主机名/IP：</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="mysql_ip_lineEdit"/>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_19">
          <property name="text">
           <string>端口号：</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="mysql_port_lineEdit"/>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_11">
          <property name="text">
           <string>用户名：</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLineEdit" name="mysql_usr_lineEdit"/>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_13">
          <property name="text">
           <string>密码</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLineEdit" name="mysql_pwd_lineEdit"/>
        </item>
       </layout>
      </widget>
      <widget class="QPushButton" name="mysql_test_pushButton">
       <property name="geometry">
        <rect>
         <x>450</x>
         <y>130</y>
         <width>75</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>测试连接</string>
       </property>
      </widget>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>501</width>
         <height>25</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QLabel" name="label_20">
          <property name="text">
           <string>临时文件目录：</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="mysql_tmpdir_lineEdit"/>
        </item>
        <item>
         <widget class="QPushButton" name="mysql_tmpdir_pushButton">
          <property name="text">
           <string>选择</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>130</y>
        <width>551</width>
        <height>71</height>
       </rect>
      </property>
      <property name="title">
       <string>HDF5存储设置</string>
      </property>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>521</width>
         <height>25</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="1">
         <widget class="QLineEdit" name="hdf5_dir_lineEdit"/>
        </item>
        <item row="0" column="2">
         <widget class="QPushButton" name="hdf5_dir_pushButton">
          <property name="text">
           <string>选择</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>目标数据（HDF5）存放目录：</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_4">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>551</width>
        <height>80</height>
       </rect>
      </property>
      <property name="title">
       <string>存储引擎设置</string>
      </property>
      <widget class="QRadioButton" name="enable_hdf55_radioButton">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>40</y>
         <width>89</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>使用HDF5</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="enable_mysql_radioButton">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>40</y>
         <width>89</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>使用MYSQL</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>执行导入</string>
     </attribute>
     <widget class="QPushButton" name="start_import_pushButton">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>20</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>执行导入</string>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox_5">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>70</y>
        <width>551</width>
        <height>451</height>
       </rect>
      </property>
      <property name="title">
       <string>导入进展</string>
      </property>
      <widget class="QWidget" name="layoutWidget_2">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>531</width>
         <height>151</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>导入日线：</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>导入分笔数据：</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_14">
          <property name="text">
           <string>导入1分钟线：</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QProgressBar" name="hdf5_trans_progressBar">
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QProgressBar" name="hdf5_day_progressBar">
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>导入分时数据：</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_12">
          <property name="text">
           <string>导入5分钟线：</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QProgressBar" name="hdf5_min_progressBar">
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QProgressBar" name="hdf5_time_progressBar">
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QProgressBar" name="hdf5_5min_progressBar">
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>导入权息数据：</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QLabel" name="hdf5_weight_label">
          <property name="text">
           <string>TextLabel</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QTextEdit" name="import_detail_textEdit">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>180</y>
         <width>531</width>
         <height>261</height>
        </rect>
       </property>
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
       <property name="html">
        <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;导入上证日线记录：&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;导入深证日线记录：&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;导入上证5分钟线记录：&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;导入深证5分钟线记录：&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;导入上证1分钟线记录：&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;导入深证1分钟线记录：&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;导入上证分笔记录：&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;导入深证分笔记录：&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;导入上证分时数据：&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;导入深证分时数据：&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;导入权息数据数：&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;导入完毕！&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
     </widget>
     <widget class="QLabel" name="import_status_label">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>20</y>
        <width>361</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>import_status_label</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>MySQL定时导入</string>
     </attribute>
     <widget class="QCheckBox" name="collect_no_week_checkBox">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>200</y>
        <width>101</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>周末不执行</string>
      </property>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>160</y>
        <width>158</width>
        <height>22</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_10">
       <item>
        <widget class="QLabel" name="label_23">
         <property name="text">
          <string>采集间隔（秒）：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="collect_sample_spinBox">
         <property name="maximum">
          <number>86400</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>30</y>
        <width>201</width>
        <height>25</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_11">
       <item>
        <widget class="QPushButton" name="collect_start_pushButton">
         <property name="text">
          <string>启动定时采集</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="collect_status_Label">
         <property name="text">
          <string>TextLabel</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>80</y>
        <width>216</width>
        <height>56</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <widget class="QLabel" name="label_21">
           <property name="text">
            <string>执行时间段1：</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <widget class="QTimeEdit" name="collect_phase1_start_timeEdit"/>
           </item>
           <item>
            <widget class="QLabel" name="label_22">
             <property name="text">
              <string>-</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QTimeEdit" name="collect_phase1_last_timeEdit"/>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_9">
         <item>
          <widget class="QLabel" name="label_24">
           <property name="text">
            <string>执行时间段2：</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_7">
           <item>
            <widget class="QTimeEdit" name="collect_phase2_start_timeEdit"/>
           </item>
           <item>
            <widget class="QLabel" name="label_25">
             <property name="text">
              <string>-</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QTimeEdit" name="collect_phase2_last_timeEdit"/>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_5">
     <attribute name="title">
      <string>控制台日志</string>
     </attribute>
     <widget class="QTextEdit" name="log_textEdit">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>571</width>
        <height>511</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
      <property name="acceptRichText">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
